pkg/service/dependencies.go

```diff
@@ -19,23 +21,31 @@ import (
 
 // Dependencies struct holds the reference to the Storage and the Manipulator interface implementations
 type Dependencies struct {
-	Storage     base.Storage
-	Manipulator Manipulator
+	Storage       base.Storage
+	Manipulator   Manipulator
+	MetricService metrics.MetricService
 }
 
 // NewDependencies constructs new Dependencies based on the config.DataSource().Kind
 // Currently, it supports only one Manipulator
-func NewDependencies() (*Dependencies, error) {
+func NewDependencies(registry *prometheus.Registry) (*Dependencies, error) {
+	var metricService metrics.MetricService
+	if regex.PrometheusMatcher.MatchString(config.MetricsSystem()) {
+		metricService = metrics.NewPrometheus(registry)
+	} else if regex.StatsdMatcher.MatchString(config.MetricsSystem()) {
+		metricService, _ = metrics.InitializeStatsdCollector(config.StatsdConfig())
+	}
+	deps := &Dependencies{Manipulator: NewManipulator(native.NewBildProcessor(), getDefaultParams(), metricService),
+		MetricService: metricService}
 	s := config.DataSource()
-	deps := &Dependencies{Manipulator: NewManipulator(native.NewBildProcessor(), getDefaultParams())}
 	if regex.WebFolderMatcher.MatchString(s.Kind) {
 		deps.Storage = NewWebFolderStorage(s.Value.(config.WebFolder), s.HystrixCommand)
 	} else if regex.S3Matcher.MatchString(s.Kind) {
 		deps.Storage = NewS3Storage(s.Value.(config.S3Bucket), s.HystrixCommand)
 	} else if regex.CloudfrontMatcher.MatchString(s.Kind) {
 		deps.Storage = NewCloudfrontStorage(s.Value.(config.Cloudfront), s.HystrixCommand)
 	}
-	if deps.Storage == nil || deps.Manipulator == nil {
+	if deps.Storage == nil || deps.Manipulator == nil || deps.MetricService == nil {
```

Comment:
`MetricService` is not an essential component, we can simply `WARN` the users that `NoOpMetricService` is being used. Blocking application initialisation here, doesn't make sense. The other thing we can do is make Prometheus the default, and we won't require `NoOpMetricService`.